cts/cli/regression.crm_resource.exp

```diff
@@ -0,0 +1,569 @@
+</pacemaker-result>
+=#=#=#= End test: Check locations and constraints for prim1 in XML - OK (0) =#=#=#=
+* Passed: crm_resource   - Check locations and constraints for prim1 in XML
+=#=#=#= Begin test: Recursively check locations and constraints for prim1 in XML =#=#=#=
+<pacemaker-result api-version="X" request="/home/clumens/src/pacemaker/tools/.libs/crm_resource -A -r prim1 --output-as=xml">
+  <locations/>
+  <status code="0" message="OK"/>
+</pacemaker-result>
+=#=#=#= End test: Recursively check locations and constraints for prim1 in XML - OK (0) =#=#=#=
+* Passed: crm_resource   - Recursively check locations and constraints for prim1 in XML
+=#=#=#= Begin test: Check locations and constraints for prim2 =#=#=#=
+Location preferences for prim2:
+  * Node cluster01 (score=INFINITY, id=prim2-on-cluster1)
+Resources prim2 is colocated with:
+  * prim3 (score=INFINITY, id=colocation-prim2-prim3-INFINITY)
+=#=#=#= End test: Check locations and constraints for prim2 - OK (0) =#=#=#=
+* Passed: crm_resource   - Check locations and constraints for prim2
+=#=#=#= Begin test: Recursively check locations and constraints for prim2 =#=#=#=
+Location preferences for prim2:
+  * Node cluster01 (score=INFINITY, id=prim2-on-cluster1)
+Resources prim2 is colocated with:
+  * prim3 (score=INFINITY, id=colocation-prim2-prim3-INFINITY)
+    * Resources prim3 is colocated with:
+      * prim4 (score=INFINITY, id=colocation-prim3-prim4-INFINITY)
+        * Location preferences for prim4:
+          * Node cluster02 (score=INFINITY, id=prim4-on-cluster2)
+        * Resources prim4 is colocated with:
+          * prim5 (score=INFINITY, id=colocation-prim4-prim5-INFINITY)
+=#=#=#= End test: Recursively check locations and constraints for prim2 - OK (0) =#=#=#=
+* Passed: crm_resource   - Recursively check locations and constraints for prim2
+=#=#=#= Begin test: Check locations and constraints for prim2 in XML =#=#=#=
+<pacemaker-result api-version="X" request="/home/clumens/src/pacemaker/tools/.libs/crm_resource -a -r prim2 --output-as=xml">
+  <locations>
+    <location host="cluster01" id="prim2-on-cluster1" score="INFINITY"/>
+  </locations>
+  <list name="rsc-is-colocated-with" count="1">
+    <colocation peer="prim3" id="colocation-prim2-prim3-INFINITY" score="INFINITY"/>
+  </list>
+  <status code="0" message="OK"/>
+</pacemaker-result>
+=#=#=#= End test: Check locations and constraints for prim2 in XML - OK (0) =#=#=#=
+* Passed: crm_resource   - Check locations and constraints for prim2 in XML
+=#=#=#= Begin test: Recursively check locations and constraints for prim2 in XML =#=#=#=
+<pacemaker-result api-version="X" request="/home/clumens/src/pacemaker/tools/.libs/crm_resource -A -r prim2 --output-as=xml">
+  <locations>
+    <location host="cluster01" id="prim2-on-cluster1" score="INFINITY"/>
+  </locations>
+  <list name="rsc-is-colocated-with" count="1">
+    <colocation peer="prim3" id="colocation-prim2-prim3-INFINITY" score="INFINITY"/>
+  </list>
```

Comment:
Recursive doesn't seem to be happening in XML

Reply:
No, it's not yet.  I was hoping to nail down the text side of things and then replicate the structure in the XML side.

Reply:
Bah, it's easy enough to fix.  Fixed, and I can always look at the other problems later.